src/ParallelAlgorithms/Interpolation/Actions/DumpInterpolatorVolumeData.hpp

```diff
@@ -0,0 +1,138 @@
+#include "Utilities/TaggedTuple.hpp"
+
+namespace intrp::Actions {
+namespace detail {
+template <typename SourceVars, typename TemporalIdTag, typename Metavariables>
+ElementVolumeData construct_element_volume_data(
+    const ElementId<Metavariables::volume_dim>& element_id,
+    const typename intrp::Tags::VolumeVarsInfo<Metavariables,
+                                               TemporalIdTag>::Info& info) {
+  std::vector<TensorComponent> components{};
+
+  const auto& all_source_vars = info.source_vars_from_element;
+  tmpl::for_each<SourceVars>(
+      [&components, &all_source_vars](auto source_var_tag_v) {
+        using source_var_tag =
+            tmpl::type_from<std::decay_t<decltype(source_var_tag_v)>>;
+        const auto& tensor = get<source_var_tag>(all_source_vars);
+        for (size_t i = 0; i < tensor.size(); i++) {
+          const auto& tensor_component = tensor[i];
+          components.emplace_back(
+              db::tag_name<source_var_tag>() + tensor.component_suffix(i),
+              tensor_component);
+        }
+      });
+
+  const auto& mesh = info.mesh;
+  const auto& index_extents = mesh.extents();
+  const auto& array_bases = mesh.basis();
+  const auto& array_quadratures = mesh.quadrature();
+  std::vector<size_t> extents(index_extents.begin(), index_extents.end());
+  std::vector<Spectral::Basis> bases(array_bases.begin(), array_bases.end());
+  std::vector<Spectral::Quadrature> quadratures(array_quadratures.begin(),
+                                                array_quadratures.end());
+  std::string element_name = MakeString{} << element_id;
+
+  return ElementVolumeData{extents, std::move(components), bases, quadratures,
+                           std::move(element_name)};
+}
+}  // namespace detail
+
+/*!
+ * \brief Dump all volume data at all temporal IDs stored in the interpolator.
+ *
+ * All dumped data will be in the usual volume files, but under different
+ * subfiles. There will be one subfile for every temporal ID type (as the
+ * interpolator can have multiple different temporal ID types from different
+ * interpolation targets). The tensors that are dumped are the ones defined in
+ * the `interpolator_source_vars` type alias in the Metavariables.
+ */
+template <typename AllTemporalIds, typename SourceVars>
+struct DumpInterpolatorVolumeData {
+  template <typename DbTagList, typename... InboxTags, typename Metavariables,
+            typename ArrayIndex, typename ActionList,
+            typename ParallelComponent>
+  static Parallel::iterable_action_return_t apply(
+      db::DataBox<DbTagList>& box,
+      const tuples::TaggedTuple<InboxTags...>& /*inboxes*/,
+      Parallel::GlobalCache<Metavariables>& cache,
+      const ArrayIndex& /*array_index*/, const ActionList /*meta*/,
+      const ParallelComponent* const /*meta*/) {
+    static_assert(
+        std::is_same_v<SourceVars,
+                       typename Metavariables::interpolator_source_vars>,
```

Comment:
Is there a need for the template parameter, rather than just using the value from the metavariables?

Reply:
I did it to avoid "reaching up" into the metavariables, which, from what I understand, we were trying to avoid. I did similar things in #4014.

Reply:
But you're reaching up into the metavariables right here.  In #4014 it was necessary for the pack expansion in the function arguments.

Reply:
Ok, then are we saying it's ok to reach up into the metavariables now? Or should I not reach up, and pass them in with the template parameter?

Reply:
My understanding (I think @nilsvu and maybe @nilsdeppe have thought about this more) is that the bad thing is that there are some things in the metavariables that have special meanings and some that are basically just local variables, and there's no easy way to tell what is what.

When a quantity is only used in a couple of places, it is usually best to just pass it in as a template parameter.

When something is used in many places and is required to be exactly the same in all of them, or is difficult to supply as a template parameter for some reason, I'm not sure if we have a clear decision about what is best.  One possibility is dedicated structs with protocols that can be pulled from the metavariables, as is done for `factory_creation` and `domain`.  Many places still do the "reaching up".

In this case, since it is already being accessed in several places, I would just use the value from the metavariables and we can fix all uses once we have decided what the correct fix is.

Reply:
I agree with your assessment @wthrowe. For the interpolation framework we should really clean up the Metavariables and put them in a struct and document them.

Reply:
Ok I just reached up for now. We can clean things up later. I did add a comment about it.